test/cpp_extensions/libtorch_agnostic_extension/libtorch_agnostic/csrc/kernel.cpp

```diff
@@ -45,7 +45,7 @@ Tensor sgd_out_of_place(
   aoti_torch_get_device_type(param.get(), &param_device_type);
 
   AtenTensorHandle out_ath;
-  aoti_torch_empty_strided(param.dim(), param_sizes, param_strides, param_dtype, param_device_type, param.get_device(), &out_ath);
+  aoti_torch_empty_strided(param.dim(), param_sizes, param_strides, param_dtype, param_device_type, param.get_device(), /*is_pinned=*/false, &out_ath);
   auto out = Tensor(out_ath);
```

Comment:
cc @desertfire  - does this api need to be bc compatible ? if so, `/is_pinned/` should go as last arg. or we can add a pinned `aoti_torch_empty_strided_pinned` api (as you did for python codegen).

Reply:
This definitely is BC-breaking. Having `is_pinned` going last is not enough. Creating `aoti_torch_empty_strided_pinned` will work, with a caveat it can cause FC-breaking, so you need to make sure call to that new function for new code path only.

Reply:
Why is it that adding the arg at the end with a default value is BC breaking ? I don't follow.